pkg/asset/manifests/infrastructure.go

```diff
@@ -93,9 +93,14 @@ func (i *Infrastructure) Generate(dependencies asset.Parents) error {
 	case baremetal.Name:
 		config.Status.PlatformStatus.Type = configv1.BareMetalPlatformType
 		config.Status.PlatformStatus.BareMetal = &configv1.BareMetalPlatformStatus{
-			APIServerInternalIP: installConfig.Config.Platform.BareMetal.APIVIP,
-			NodeDNSIP:           installConfig.Config.Platform.BareMetal.DNSVIP,
-			IngressIP:           installConfig.Config.Platform.BareMetal.IngressVIP,
+			APIServerInternalIP:     installConfig.Config.Platform.BareMetal.APIVIP,
+			NodeDNSIP:               installConfig.Config.Platform.BareMetal.DNSVIP,
+			IngressIP:               installConfig.Config.Platform.BareMetal.IngressVIP,
+			ProvisioningInterface:   installConfig.Config.Platform.BareMetal.ProvisioningInterface,
+			ProvisioningNetworkCIDR: installConfig.Config.Platform.BareMetal.ProvisioningNetworkCIDR,
+			ProvisioningIP:          installConfig.Config.Platform.BareMetal.ProvisioningIP,
+			ProvisioningDHCPRange:   installConfig.Config.Platform.BareMetal.ProvisioningDHCPRange,
+			RhcosImageURL:           installConfig.Config.Platform.BareMetal.RhcosImageURL,
```

Comment:
The RHCOS image URL shouldn't be provided via the installConfig, it's generated from the `data/data/rhcos.json` - e.g see https://github.com/openshift/installer/commit/ac890748ade6cf46b8bd99bcf62568d412c5f4a7

Reply:
To add more detail, I think you'll need to use a similar approach to https://github.com/openshift/installer/blob/master/pkg/asset/cluster/tfvars.go#L77 where `"github.com/openshift/installer/pkg/asset/rhcos"` gets imported, then you add `new(rhcos.Image),` to the Dependencies of this asset, and to the `parents.Get` in the Generate function